src/js/angular/core/directives.js

```diff
@@ -864,3 +865,32 @@ function inactivePluginDirective(toastr, RDF4JRepositoriesRestService, ModalServ
         checkPluginIsActive();
     }
 }
+
+function autoGrowDirective() {
+    return {
+        restrict: 'A',
+        require: 'ngModel',
+        link: function(scope, element, attrs, ngModelCtrl) {
+            const updateHeight = function() {
+                const value = ngModelCtrl.$viewValue;
+                if (!value || value.trim() === '') {
+                    element[0].style.height = '26px';
+                } else {
+                    element[0].style.height = '0px'; // Reset height to calculate new scrollHeight
+                    element[0].style.height = element[0].scrollHeight + 'px';
+                }
+                element[0].style.overflow = 'hidden';
+            };
+
+            scope.$watch(function() {
```

Comment:
when call $watch it returns unsubscribe function. I think we have to call it on destroy.

Reply:
+1